File: lib/BW_Simulation_API/BW_PythonAPI/API_Example.py
import BWSimREST
import time
print BWSimREST.addClient('Client1','MySensor')
print BWSimREST.addNewSubscribe('Client1','Topic')
print BWSimREST.addNewSubscribe('Client1','Topic1')
print BWSimREST.addClient('Client2','Mobile')
print BWSimREST.addNewSubscribe('Client2','Topic')
print BWSimREST.addNewSubscribe('Client2','Topic2')
print BWSimREST.addClient('Client3','NewSensor')
print BWSimREST.addNewSubscribe('Client3','Topic')
print BWSimREST.addNewSubscribe('Client3','Topic3')
print BWSimREST.addResponseforRequest('Client1','Topic','Hello','Topic1','ReqRes1')
print BWSimREST.addResponseforRequest('Client2','Topic','Hello','Topic2','ReqRes2')
print BWSimREST.addResponseforRequest('Client3','Topic','Hello','Topic3','ReqRes3')
print BWSimREST.startclient('Client1')
print BWSimREST.startclient('Client2')
print BWSimREST.startclient('Client3')
time.sleep(5)
print BWSimREST.publish('pubclient1','Topic','Hello')
time.sleep(3)
print BWSimREST.publish('pubclient2','Topic','Hello')
time.sleep(4)
print 'Unsubscribing Topic1,Topic2,Topic3'
print BWSimREST.unsubscribe('Client1','Topic1')
print BWSimREST.unsubscribe('Client2','Topic2')
print BWSimREST.unsubscribe('Client3','Topic3')
time.sleep(5)
print 'Publishing Topic'
print BWSimREST.publish('pubclient1','Topic','Hello')
print BWSimREST.publish('pubclient2','Topic','Hello')

File: lib/BW_Simulation_API/BW_JavaAPI/API_Example.java
import BW_JavaAPI.BWSimREST;
class API_Example
{
	public static void main(String arg[]) throws InterruptedException
	{
		BWSimREST api=new BWSimREST();
		String s;
		s=api.addClient("Client1","MySensor");
		System.out.println(s);
		s=api.addNewSubscribe("Client1","Topic");
		System.out.println(s);
		s=api.addNewSubscribe("Client1","Topic1");
		System.out.println(s);
		s=api.addClient("Client2","Mobile");
		System.out.println(s);
		s=api.addNewSubscribe("Client2","Topic");
		System.out.println(s);
		s=api.addNewSubscribe("Client2","Topic2");
		System.out.println(s);
		s=api.addClient("Client3","NewSensor");
		System.out.println(s);
		s=api.addNewSubscribe("Client3","Topic");
		System.out.println(s);
		s=api.addNewSubscribe("Client3","Topic3");
		System.out.println(s);
		s=api.addResponseforRequest("Client1","Topic","Hello","Topic1","ReqRes1");
		System.out.println(s);
		s=api.addResponseforRequest("Client2","Topic","Hello","Topic2","ReqRes2");
		System.out.println(s);
		s=api.addResponseforRequest("Client3","Topic","Hello","Topic3","ReqRes3");
		System.out.println(s);
		s=api.startclient("Client1");
		System.out.println(s);
		s=api.startclient("Client2");
		System.out.println(s);
		s=api.startclient("Client3");
		Thread.sleep(10000);
		s=api.publish("pubclient1","Topic","Hello");
		Thread.sleep(5000);
		s=api.publish("pubclient2","Topic","Hello");
		Thread.sleep(3000);
		System.out.println("Unsubscribing Topic1,Topic2,Topic3");
		s=api.unsubscribe("Client1","Topic1");
		System.out.println(s);
		s=api.unsubscribe("Client2","Topic2");
		System.out.println(s);
		s=api.unsubscribe("Client3","Topic3");
		System.out.println(s);
		System.out.println("Publishing Topic");
		s=api.publish("pubclient1","Topic","Hello");
		System.out.println(s);
		s=api.publish("pubclient2","Topic","Hello");
		System.out.println(s);
		Thread.sleep(2000);
		s=api.stopclient("Client1");
		System.out.println(s);
		s=api.stopclient("Client2");
		System.out.println(s);
		s=api.stopclient("Client3");
		System.out.println(s);


	}
}
